scripts_src/global/gl_k_alcohl.ssl

```diff
@@ -62,24 +70,38 @@ procedure drink(variable who, variable item) begin
     if (global_var(GVAR_ALCOHOL_ADDICT) == 1) then return;
 
     // regular drinking increases chances
-    if (game_time - last_drink_date >= TIME_ONE_WEEK)
-        then chance = 1;
-    else if (game_time - last_drink_date >= TIME_THREE_DAYS)
-        then chance = 5;
-    else chance = 10;
+    if ((game_time - last_drink_date) >= SPORADIC_ALCOHOL_USAGE_TIME) then begin
+        chance = 1;
+//        display_msg("<test> last drink longer than a week ago. Your base chance of addiction is : "+chance);
+    end
+    else if ((game_time - last_drink_date) >= MEDIUM_ALCOHOL_USAGE_TIME) then begin
+        chance = 5;
+//        display_msg("<test> last drink longer than 3 days ago. Your base chance of addiction is : "+chance);
+    end
+    else begin
+        chance = 10;
+//        display_msg("<test> last drink less than 3 days ago. Your base chance of addiction is : "+chance);
+    end
     roll = random(1, 100);
+//    display_msg("<test> the addiction roll was: "+roll);
 
     // Drug reliant: double chance
     if has_trait(TRAIT_TRAIT, dude_obj, TRAIT_drug_addict) then chance = chance * 2;
     // Drug resistant: half chance
     if has_trait(TRAIT_TRAIT, dude_obj, TRAIT_drug_resistant) then chance = chance * 0.5;
+    if is_booze(item) then chance = chance * 2;
```

Comment:
I don't think strong alcohol is more addictive, actually, it's more the regularity of it.

Reply:
Well without this change drinking two beers directly one after the other has a way higher addiction chance than drinking one bottle of booze, even though that bottle of booze has more alcohol in it and the time of consumption is more or less the same.
(Consuming two beers has almost a 6% chance to get addicted, consuming one booze has just 1%)

But I really don't feel all that strongly about it, I'll just revert the patch then. 🤷

Reply:
I think good approximation would be arctangent function, on both daily and weekly level, but like I said, too lazy to implement it properly. For now can just leave it like that, maybe add extra features later.